core/src/main/java/io/questdb/Bootstrap.java

```diff
@@ -0,0 +1,475 @@
+        File publicFolder = f.getParentFile();
+        if (!publicFolder.exists()) {
+            if (!publicFolder.mkdirs()) {
+                throw new BootstrapException("Cannot create folder: " + publicFolder);
+            }
+        }
+        try (FileOutputStream fos = new FileOutputStream(f)) {
+            byte[] buf = version.getBytes();
+            fos.write(buf, 0, buf.length);
+        }
+    }
+
+    private void reportValidateConfig() {
+        final boolean httpEnabled = config.getHttpServerConfiguration().isEnabled();
+        final boolean httpReadOnly = config.getHttpServerConfiguration().getHttpContextConfiguration().readOnlySecurityContext();
+        final String httpReadOnlyHint = httpEnabled && httpReadOnly ? " [read-only]" : "";
+        final boolean pgEnabled = config.getPGWireConfiguration().isEnabled();
+        final boolean pgReadOnly = config.getPGWireConfiguration().readOnlySecurityContext();
+        final String pgReadOnlyHint = pgEnabled && pgReadOnly ? " [read-only]" : "";
+        final CairoConfiguration cairoConfiguration = config.getCairoConfiguration();
+        log.advisoryW().$("Config changes applied:").$();
+        log.advisoryW().$("  http.enabled : ").$(httpEnabled).$(httpReadOnlyHint).$();
+        log.advisoryW().$("  tcp.enabled  : ").$(config.getLineTcpReceiverConfiguration().isEnabled()).$();
+        log.advisoryW().$("  pg.enabled   : ").$(pgEnabled).$(pgReadOnlyHint).$();
+        log.advisoryW().$("open database [id=").$(cairoConfiguration.getDatabaseIdLo()).$('.').$(cairoConfiguration.getDatabaseIdHi()).$(']').$();
+        log.advisoryW().$("platform [bit=").$(System.getProperty("sun.arch.data.model")).$(']').$();
+        switch (Os.type) {
+            case Os.WINDOWS:
+                log.advisoryW().$("OS/Arch: windows/amd64").$(Vect.getSupportedInstructionSetName()).$();
+                break;
+            case Os.LINUX_AMD64:
+                log.advisoryW().$("OS/Arch: linux/amd64").$(Vect.getSupportedInstructionSetName()).$();
+                break;
+            case Os.OSX_AMD64:
+                log.advisoryW().$("OS/Arch: apple/amd64").$(Vect.getSupportedInstructionSetName()).$();
+                break;
+            case Os.OSX_ARM64:
+                log.advisoryW().$("OS/Arch: apple/apple-silicon").$();
+                break;
+            case Os.LINUX_ARM64:
+                log.advisoryW().$("OS/Arch: linux/arm64").$(Vect.getSupportedInstructionSetName()).$();
+                break;
+            case Os.FREEBSD:
+                log.advisoryW().$("OS: freebsd/amd64").$(Vect.getSupportedInstructionSetName()).$();
+                break;
+            default:
+                log.criticalW().$("Unsupported OS").$(Vect.getSupportedInstructionSetName()).$();
+                break;
+        }
+        log.advisoryW().$("available CPUs: ").$(Runtime.getRuntime().availableProcessors()).$();
+        log.advisoryW().$("db root: ").$(cairoConfiguration.getRoot()).$();

```

Comment:
nit: we missed adding SQL copy root and work dir here and also `verifyFileSystem()` calls for them (when they're set). Might be a good idea to add them within this PR.

Reply:
will do